APIImpl/PluginManager.lua

```diff
@@ -29,17 +42,118 @@ end
 
 
 
+--- Implementation of the cPluginManager:BindConsoleCommand function
+-- There are two separate API endpoints for the same implementation, so the implementation is pulled into a common function
+local function BindConsoleCommand(a_Simulator, a_Self, a_Command, a_Callback, a_HelpString)
+	if (a_Command:match("%s")) then
+		a_Simulator.logger:trace("Registering a console command containing whitespace(%s), such a handler will never be called by Cuberite.%s", a_Command, debug.traceback("", 2))
+	end
+	if (a_Simulator.registeredConsoleCommandHandlers[a_Command]) then
+		a_Simulator.logger:info("Registering  a console command that is already registered (%s). The second handler will not be called / tested.", a_Command)
+		return
+	end
+	a_Simulator.registeredConsoleCommandHandlers[a_Command] =
+	{
+		callback = a_Callback,
+		helpString = a_HelpString,
+	}
+	return true
+end
+
+
+
+
+
+--- Implementation of the cPluginManager:ForEachPlugin function
+-- There are two separate API endpoints for the same implementation
+local function ForEachPlugin(a_Simulator, a_Self, a_Callback)
+	local res = a_Simulator:processCallbackRequest({
+		Function = a_Callback,
+		ParamValues = { a_Self:GetCurrentPlugin() },
+		Notes = "cPluginManager:ForEachPlugin() callback",
+	})
+	return (res == true) and false or true  -- If the callback signalized an abort, return false
+end
+
+
+
+
+
 return
 {
-	["<static> cPluginManager:AddHook(cPluginManager#PluginHook, function)"] = function (a_Simulator, a_Self, a_HookType, a_Callback)
+	["cPlugin:GetLocalFolder()"] = function(a_Simulator, a_Self)
+		return getmetatable(a_Self).simulatorInternal_pluginPath or a_Simulator:createInstance({Type = "string"})
+	end,
+
+	["cPluginLua:AddWebTab(string, function)"] = function(a_Simulator, a_Self, a_Title, a_Callback)
+		a_Simulator.logger:warning("Plugin uses the deprecated function cPluginLua:AddWebTab")
```

Comment:
The flag should not only specify that it IS deprecated, it should also specify what to use instead. And APIDump should process this information into the documentation.